tests/regression_tests/transient_mg/test.py

```diff
@@ -0,0 +1,303 @@
+    scatter_matrix = np.rollaxis(scatter_matrix, 0, 3)
+    h2o_xsdata.set_scatter_matrix(scatter_matrix)
+    h2o_xsdata.set_inverse_velocity([1./2.23466E+09, 1./5.07347E+08, 1./3.86595E+07,
+        1./5.13931E+06, 1./1.67734E+06, 1./7.28603E+05, 1./2.92902E+05])
+
+    mg_cross_sections_file = openmc.MGXSLibrary(groups, 8)
+    mg_cross_sections_file.add_xsdatas([uo2_xsdata, h2o_xsdata])
+    mg_cross_sections_file.export_to_hdf5()
+
+    uo2_data = openmc.Macroscopic('UO2')
+    h2o_data = openmc.Macroscopic('LWTR')
+
+    uo2 = openmc.Material(name='UO2 fuel')
+    uo2.set_density('macro', 1.0)
+    uo2.add_macroscopic(uo2_data)
+
+    water = openmc.Material(name='Water')
+    water.set_density('macro', 1.0)
+    water.add_macroscopic(h2o_data)
+
+    model.materials = openmc.Materials([uo2, water])
+    model.materials.cross_sections = "mgxs.h5"
+
+    fuel_or = openmc.ZCylinder(r=0.54, name='Fuel OR')
+
+    pitch = 1.26
+    box = openmc.rectangular_prism(pitch, pitch, boundary_type='reflective')
+    bottom = openmc.ZPlane(z0=-10, boundary_type='vacuum')
+    top = openmc.ZPlane(z0= 10, boundary_type='vacuum')
+
+    fuel = openmc.Cell(fill=uo2, region=-fuel_or & +bottom & -top, name='fuel')
+    moderator = openmc.Cell(fill=water, region=+fuel_or & box & +bottom & -top, name='moderator')
+
+    model.geometry = openmc.Geometry([fuel, moderator])
+    
+    model.settings.batches = 100
+    model.settings.inactive = 10
+    model.settings.particles = 100
+    model.settings.output = {'tallies': False}
+    model.settings.energy_mode = 'multi-group'
+
+    lower_left = [-0.63, -0.63, -10]
+    upper_right = [0.63, 0.63, 10]
+    uniform_dist = openmc.stats.Box(lower_left, upper_right, only_fissionable=True)
+    model.settings.source = openmc.source.Source(space=uniform_dist)
+
+    sourcepoint = dict()
+    sourcepoint['batches'] = [model.settings.batches]
+    sourcepoint['separate'] = True
+    sourcepoint['write'] = True
+    model.settings.sourcepoint = sourcepoint
+
+    full_pin_cell_mesh = openmc.RegularMesh()
+    full_pin_cell_mesh.dimension = [1,1,20]
+    full_pin_cell_mesh.lower_left = [-0.63,-0.63,-10]
+    full_pin_cell_mesh.width =[0.63*2,0.63*2,1.0]
+
+    energy_groups = openmc.mgxs.EnergyGroups()
+    energy_groups.group_edges = [0, 0.0635, 10.0, 1.0e2, 1.0e3, 0.5e6, 1.0e6, 20.e6]
+
+    one_group = openmc.mgxs.EnergyGroups()
+    one_group.group_edges = [0, 20.e6]
+
+    t_outer = np.arange(0., 1.0, 5.e-1)
+    clock = openmc.kinetics.Clock(dt_inner=1.e-2, t_outer=t_outer)
+
+    transient = {}
+    for material in model.materials:
+        MatChange = {material.name: {},}
+        transient.update(MatChange)
+        for t in t_outer:
+            time_dict = {
+            t: {
+                'density' : {},
+                'temperature' : {},
+                }
+            }
+            transient[material.name].update(time_dict)
```

Comment:
I think this is equivalent?
```suggestion
            transient[material.name][t] = {
                'density': {},
                'temperature': {}
            }
```